users/postgres/users.go

```diff
@@ -62,15 +79,19 @@ func (ur userRepository) Update(ctx context.Context, user users.User) error {
 func (ur userRepository) UpdateUser(ctx context.Context, user users.User) error {
 	q := `UPDATE users SET metadata = :metadata WHERE email = :email`
 
-	dbu := toDBUser(user)
+	dbu, err := toDBUser(user)
+	if err != nil {
+		return errors.Wrap(errUpdateUserDB, err)
+	}
 	if _, err := ur.db.NamedExecContext(ctx, q, dbu); err != nil {
 		return errors.Wrap(errUpdateUserDB, err)
 	}
 
 	return nil
 }
 
-func (ur userRepository) RetrieveByEmail(ctx context.Context, email string) (users.User, error) {
+func (ur userRepository) RetrieveByEmail(ctx context.Context, email string, groups bool) (users.User, error) {
```

Comment:
Why this group retrieves? Do we need this?

Reply:
Probably not, that was my initial idea to have a flag whether you want to retrieve all groups that user belongs along
but I made a methods `RetrieveAllGroupsForUser` and `RetrieveAllUsersForGroup`

Reply:
I see. Yes, I also think that we don't need it now, its better to decouple.